service/csv/pokemon.go

```diff
@@ -0,0 +1,225 @@
+package service_csv
+
+import (
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/rayl17/academy-go-q42021/model"
+
+	parameters "github.com/rayl17/academy-go-q42021/global"
+)
+
+type ServiceInterface interface {
+	GetPokemonByName(name string) (*model.Pokemon, error)
+	GetPokemonByID(id string) (*model.Pokemon, error)
+	getPokemons() []model.Pokemon
+	SearchPokemonApi(name string) (string, error)
+	GetPokemonsByTypes(types string, items int, items_per_worker int) (*[]model.Pokemon, error)
+}
+
+type PokemonService struct {
+	servicePath string
+}
+
+func NewPokemonService(path string) (*PokemonService, error) {
+
+	return &PokemonService{
+		servicePath: path,
+	}, nil
+}
+
+// this method recives a name and return the pokemon with that name
+func (ps *PokemonService) GetPokemonByName(name string) (*model.Pokemon, error) {
+
+	for _, pokemon := range ps.getPokemons() {
+		if pokemon.Name == name {
+			return &pokemon, nil
+		}
+	}
+
+	return nil, errors.New("Pokemon not found ")
+}
+
+/// This method recives a id to look for and returns the pokemon with that id ,
+func (ps *PokemonService) GetPokemonByID(id string) (*model.Pokemon, error) {
+
+	for _, pokemon := range ps.getPokemons() {
+		if pokemon.Id == id {
+			return &pokemon, nil
+		}
+	}
+
+	return nil, errors.New("Pokemon not found")
+}
+
+// this fmethod reads thecvs file and   return a list of pokemons correctli formated  from the csv file
+func (p *PokemonService) getPokemons() []model.Pokemon {
+	f, err := os.Open(p.servicePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	defer f.Close()
+
+	csvReader := csv.NewReader(f)
+	data, err1 := csvReader.ReadAll()
+
+	if err1 != nil {
+		log.Fatal(err)
+	}
+
+	pokemons := formatPokemonList(data)
+	return pokemons
+}
+
+/// this mehod  make a get request to the api , call the savePokemonCSV function and return the response
+/// if is succesfully or not
+func (ps *PokemonService) SearchPokemonApi(name string) (string, error) {
```

Comment:
Acronyms all UPPERCASE.
```suggestion
// SearchPokemonAPI make a get request to the API, call the savePokemonCSV function and return the response
// if is succesfully or not
func (ps *PokemonService) SearchPokemonAPI(name string) (string, error) {
```